### PR TITLE
improve loading of FFI for rubinius support

### PR DESCRIPTION
**Pull request by abargnesi**
tested loading FFI / librdf under rubinius, jruby, and mri

added integration spec to ensure loading FFI and librdf does not raise
errors

added travis-ci configuration to run test suite on mri, jruby, and
rubinius

### CODE REVIEW

**Comment by abargnesi** [comment]
The test suite is still failing in relation to issue #3.

I added a travis build to run against different ruby versions / engines. This can only raise by motivation to fix the pointer errors I'm getting in redlander that I do not get with the ruby redland-bindings.

**Comment by abargnesi** [comment]
travis build success running just `spec/integration/ruby_engine_spec.rb`: https://travis-ci.org/abargnesi/redlander/builds/31244142

travis build failure in relation to #3: https://travis-ci.org/abargnesi/redlander/builds/31266547

File: lib/redland.rb [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+        extend FFI::Library
+      end
+    end
+    # @api_private
+    # Loads the librdf shared library.
+    def load_librdf
+      load_ffi
+
+      begin
+        ffi_lib "rd"
```
Comment (cordawyn) [inline lib/redland.rb:31]: Was it meant to be `ffi_lib "rdf"` here?

**Comment by cordawyn** [comment]
@abargnesi Great! Just let's sort it out with the above remark, and I'll merge your pull request right away.

**Comment by abargnesi** [comment]
@cordawyn Yes that is a typo.

There is another issue to address with rubinius (tested 2.2.10) where we refer to `FFI::Pointer` when it should be `Rubinius::FFI::Pointer`. We can address this by adding an `ffi` reader class method on the Redland module that returns the correct module for the platform.

I will submit another pull request for this later today.

**Comment by cordawyn** [comment]
You could've updated this pull request instead ;-) But I'm okay with a new one too.

**Pull request closed without merging**